### PR TITLE
fix(query-bar): incorrect text on ColumnButton after hiding

### PR DESCRIPTION
**Pull request by ChenCodes**
Changes in this PR:
- When one column is hidden, show `1 Column Hidden` on the Column Button instead of `1 Columns Hidden`

Screen Shot 2019-04-04 at 3 40 36 PM

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @ChenCodes has signed the CLA. Thanks for the pull request!

File: i18n/en-US.properties [hunk 1]
```diff
@@ -808,6 +808,8 @@ boxui.presence.timeSinceLastPreviewed = Previewed {timeAgo}
 boxui.queryBar.addFilterButtonText = + Add Filter
 # Text on the apply filter button, on click applies the filters
 boxui.queryBar.applyFiltersButtonText = Apply
+# Text on the columns button, if one column has been hidden then it will display this text
+boxui.queryBar.columnHiddenButtonText = {number} Column Hidden
```
Comment (macrotim) [inline i18n/en-US.properties]: Wouldn't you replace {number} with 1?

File: src/features/query-bar/__tests__/ColumnButton-test.js [hunk 2]
```diff
@@ -28,6 +28,26 @@ describe('features/query-bar/components/ColumnButton', () => {
             const wrapper = getWrapper({ template });
             expect(wrapper).toMatchSnapshot();
         });
+
+        test('should render ColumnButton with correct message when all columns are visible', () => {
```
Comment (macrotim) [inline src/features/query-bar/__tests__/ColumnButton-test.js]: test.each?

File: src/features/query-bar/components/ColumnButton.js [hunk 3]
```diff
@@ -67,6 +67,28 @@ class ColumnButton extends React.Component<Props, State> {
 
         const buttonClasses = classNames('query-bar-button', numberOfHiddenColumns !== 0 ? 'is-active' : '');
 
+        let columnsButtonText;
+        let message;
+        switch (true) {
```
Comment (macrotim) [inline src/features/query-bar/components/ColumnButton.js]: ?

Please switch to if().

File: src/features/query-bar/__tests__/ColumnButton-test.js [hunk 4]
```diff
@@ -28,6 +27,21 @@ describe('features/query-bar/components/ColumnButton', () => {
             const wrapper = getWrapper({ template });
             expect(wrapper).toMatchSnapshot();
         });
+
+        const allVisibleColumns = [columnForDateType, columnWithFloatType];
```
Comment (patrickpaul) [inline src/features/query-bar/__tests__/ColumnButton-test.js]: -> `visibleColumns`

File: i18n/en-US.properties [hunk 5]
```diff
@@ -808,6 +808,8 @@ boxui.presence.timeSinceLastPreviewed = Previewed {timeAgo}
 boxui.queryBar.addFilterButtonText = + Add Filter
 # Text on the apply filter button, on click applies the filters
 boxui.queryBar.applyFiltersButtonText = Apply
+# Text on the columns button, if one column has been hidden then it will display this text
+boxui.queryBar.columnHiddenButtonText = 1 Column Hidden
```
Comment (patrickpaul) [inline i18n/en-US.properties]: We don't need multiple messages... make use of `react-intl`. The unified message could look like:

`{count, plural, =1 {Column} other {Columns} Hidden`

**Pull request merged**